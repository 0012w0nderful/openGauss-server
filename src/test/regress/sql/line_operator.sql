select ?- line '(1,2),(3,2)';
select ?| line '(1,2),(3,2)';
select line '(1,2),(3,2)' ## lseg '(1,2),(3,2)';
select line '(1,2),(3,2)' ## box '(1,1),(2,2)';
select ## line '(1,2),(3,2)';
select ## line '(1,2),(3,2)';
select line '(1,2),(3,2)' <-> box '(1,1),(2,2)';
select point '1,2' <-> line '(1,2),(3,2)';
select lseg '(1,2),(3,2)' <->  line '(1,2),(3,2)';
select line '(1,2),(3,2)' <->  line '(1,2),(3,2)';
select line '(1,2),(3,2)' = line '(1,2),(3,2)';
select line '(1,2),(3,2)' ?# line '(1,2),(3,2)';
select line '(1,2),(3,2)' # line '(1,2),(3,2)';
select line '(1,2),(3,2)' ?|| line '(1,2),(3,2)';
select line '(1,2),(3,2)' ?-| line '(1,2),(3,2)';
select line '(1,2),(3,2)' ?# box '(1,1),(2,2)';
select lseg '(1,2),(3,2)' ?# line '(1,2),(3,2)';
select lseg '(1,2),(3,2)' @ line '(1,2),(3,2)';
select lseg '(1,2),(3,2)' <@line '(1,2),(3,2)';
select point '1,2' @ line '(1,2),(3,2)';
select point '1,2' <@line '(1,2),(3,2)';
