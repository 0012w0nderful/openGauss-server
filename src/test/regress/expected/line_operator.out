select ?- line '(1,2),(3,2)';
 ?column? 
----------
 t
(1 row)

select ?| line '(1,2),(3,2)';
 ?column? 
----------
 f
(1 row)

select line '(1,2),(3,2)' ## lseg '(1,2),(3,2)';
 ?column? 
----------
 (3,2)
(1 row)

select line '(1,2),(3,2)' ## box '(1,1),(2,2)';
ERROR:  function "close_lb" not implemented
select ## line '(1,2),(3,2)';
ERROR:  operator does not exist: ## line
LINE 1: select ## line '(1,2),(3,2)';
               ^
HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.
select ## line '(1,2),(3,2)';
ERROR:  operator does not exist: ## line
LINE 1: select ## line '(1,2),(3,2)';
               ^
HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.
select line '(1,2),(3,2)' <-> box '(1,1),(2,2)';
ERROR:  function "dist_lb" not implemented
select point '1,2' <-> line '(1,2),(3,2)';
 ?column? 
----------
        0
(1 row)

select lseg '(1,2),(3,2)' <->  line '(1,2),(3,2)';
 ?column? 
----------
        0
(1 row)

select line '(1,2),(3,2)' <->  line '(1,2),(3,2)';
 ?column? 
----------
        0
(1 row)

select line '(1,2),(3,2)' = line '(1,2),(3,2)';
 ?column? 
----------
 t
(1 row)

select line '(1,2),(3,2)' ?# line '(1,2),(3,2)';
 ?column? 
----------
 f
(1 row)

select line '(1,2),(3,2)' # line '(1,2),(3,2)';
 ?column? 
----------
 
(1 row)

select line '(1,2),(3,2)' ?|| line '(1,2),(3,2)';
 ?column? 
----------
 t
(1 row)

select line '(1,2),(3,2)' ?-| line '(1,2),(3,2)';
 ?column? 
----------
 f
(1 row)

select line '(1,2),(3,2)' ?# box '(1,1),(2,2)';
 ?column? 
----------
 t
(1 row)

select lseg '(1,2),(3,2)' ?# line '(1,2),(3,2)';
 ?column? 
----------
 f
(1 row)

select lseg '(1,2),(3,2)' @ line '(1,2),(3,2)';
 ?column? 
----------
 t
(1 row)

select lseg '(1,2),(3,2)' <@line '(1,2),(3,2)';
 ?column? 
----------
 t
(1 row)

select point '1,2' @ line '(1,2),(3,2)';
 ?column? 
----------
 t
(1 row)

select point '1,2' <@line '(1,2),(3,2)';
 ?column? 
----------
 t
(1 row)

